### PR TITLE
NMS-13408: Document search panel

### PR DESCRIPTION
**Pull request by Bonrob2**
NMS-13408: Document search panel - added documentation for search panel and metadata in search feature. Resubmitting PR due to issue with out of sync branch.

### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/${JIRA-ISSUE-NUMBER}

### CODE REVIEW

**Comment by Bonrob2** [comment]
Creating a draft PR, as the release 28.x branch did not have updates from foundation-2021 that fixed the variables and heading issues in a few files. Once content here is approved, I'll update once Ben's merge forward has completed and re-submit proper PR.

File: docs/modules/operation/pages/search/search.adoc
Comment (christianpape) [inline docs/modules/operation/pages/search/search.adoc:13]: That's not correct. The search results do only contain metadata information for the matches found in metadata. For instance if you search for 'foo' and you find a node due to a match in its name 'foobar' no metadata is displayed at all. But if you search for 'foo' and a match is found for a node 'bar' in its metadata customkey='somefoo' it is displayed. So, the idea for the feature "Customizing search results" below, was to give an user an option to define at least on line of information, that will be displayed for each node regardless how the match was found.

**Review by christianpape — CHANGES_REQUESTED** [review]

File: docs/modules/operation/pages/search/search.adoc
Comment (Bonrob2) [reply docs/modules/operation/pages/search/search.adoc:13]: Thank you, @christianpape for clarifying that. I will update.

File: docs/modules/operation/pages/search/search.adoc
Comment (indigo423) [inline docs/modules/operation/pages/search/search.adoc:9]: We can use the Keyboard macro `btn:[Shift]`.

**Review by indigo423 — APPROVED** [review]
Besides the comment from @christianpape all good to me. Nice!

File: docs/modules/operation/pages/search/search.adoc
Comment (Bonrob2) [reply docs/modules/operation/pages/search/search.adoc]: I learn something new everyday! Thanks updated.

**Review by christianpape — APPROVED** [review]
LGTM.

**Pull request merged**
